### PR TITLE
Add Box::downcast() for dyn Any + Send + Sync

### PR DESCRIPTION
**Pull request by sdroege**
Looks like a plain omission, but unfortunately I just needed that in my code :)

### CODE REVIEW

File: library/alloc/src/boxed.rs [hunk 1]
```diff
@@ -1364,6 +1364,38 @@ impl<A: Allocator> Box<dyn Any + Send, A> {
     }
 }
 
+impl<A: Allocator> Box<dyn Any + Send + Sync, A> {
+    #[inline]
+    #[stable(feature = "XXX", since = "1.51.0")]
```
Comment (sdroege) [inline library/alloc/src/boxed.rs:1369]: What should be done about the feature name here?

Comment (GuillaumeGomez) [reply library/alloc/src/boxed.rs:1369]: What do you think about "box_send_sync_any_downcast"?

Comment (sdroege) [reply library/alloc/src/boxed.rs:1369]: Thanks, let's go with that then :)

**Comment by KodrAus** [comment]
This _could_ be added as unstable first, but that doesn’t seem necessary to me.

**Comment by sdroege** [comment]
> @rfcbot fcp merge

@KodrAus I believe this did not work, maybe because the T-libs label was added later? Or at least it does not show up on rfcbot.rs

**Comment by GuillaumeGomez** [comment]
ping @rust-lang/libs , the fcg-merge instruction failed I think?

**Comment by m-ou-se** [comment]
@KodrAus in https://github.com/rust-lang/rust/pull/80945#issuecomment-760203478:
> This does just seem like an omission to me.
> 
> @rfcbot fcp merge
> 
> This adds `Box<dyn Any + Send + Sync>::downcast`, which already exists for `Box<dyn Any>` and `Box<dyn Any + Send>`.

**Comment by m-ou-se** [comment]
@KodrAus in https://github.com/rust-lang/rust/pull/80945#issuecomment-760203478:
> This does just seem like an omission to me.
> 
> @rfcbot fcp merge
>
> This adds `Box<dyn Any + Send + Sync>::downcast`, which already exists for `Box<dyn Any>` and `Box<dyn Any + Send>`.

@rfcbot fcp merge

**Pull request merged**